tree/node.go

```diff
@@ -0,0 +1,60 @@
+package tree
+
+import (
+	"fmt"
+	"sort"
+)
+
+// Node is a tree of string. Leaves are empty Nodes or nil.
+type Node map[string]Node
+
+// Print recurses through n and its member Nodes to pretty-print a Node as a
+// tree such as:
+//
+//     ├── ftree
+//     │   ├── node.go
+//     │   ├── tree.go
+//     │   └── tree_test.go
+//     └── main.go
+//
+// Print should initially be called as n.Print(0, nil). The arguments are
+// primarily interesting during recursion.
+func (n Node) Print(indent int, leaders []string) {
+	var keys []string
+	for k := range n {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+
+	for i, part := range keys {
+		if part == "" {
+			n[part].Print(indent, leaders)
+			continue
+		}
+
+		for _, l := range leaders {
```

Comment:
What do you think about expanding `l` into something more descriptive, i.e `leader`?